### PR TITLE
[Draft] fix some parser compatibility for cypher

### PR DESCRIPTION
**Pull request by czpmango**
Modify some cypher symbol compatibility to improve the ease of LDBC testing.
Besides, fix some bison potential Shift/Reduce conflicts.
#### What type of PR is this?
- [ ] bug
- [ ] feature
- [ ] enhancement

#### Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [x] Documentation affected (Please add the label if documentation needs to be modified.)
- [x] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

### CODE REVIEW

**Review by Aiee — APPROVED** [review]

**Review by jievince — DISMISSED** [review]

File: src/parser/parser.yy [hunk 1]
```diff
@@ -170,21 +170,21 @@ static constexpr size_t kCommentLengthLimit = 256;
...
-%token KW_GET KW_DECLARE KW_GRAPH KW_META KW_STORAGE KW_AGENT
-%token KW_TTL KW_TTL_DURATION KW_TTL_COL KW_DATA KW_STOP
+%token KW_GET KW_GRAPH KW_META KW_STORAGE KW_AGENT
+%token KW_TTL_DURATION KW_TTL_COL KW_DATA KW_STOP
 %token KW_FETCH KW_PROP KW_UPDATE KW_UPSERT KW_WHEN
 %token KW_ORDER KW_ASC KW_LIMIT KW_SAMPLE KW_OFFSET KW_ASCENDING KW_DESCENDING
 %token KW_DISTINCT KW_ALL KW_OF
 %token KW_BALANCE KW_LEADER KW_RESET KW_PLAN
 %token KW_SHORTEST KW_PATH KW_NOLOOP
-%token KW_IS KW_NULL KW_DEFAULT
```
Comment (CPWstatic) [inline src/parser/parser.yy:187]: I thought `IS` is now used as key word?

Comment (jievince) [reply src/parser/parser.yy:187]: Not used. The keywords are `KW_IS_NULL`, `KW_IS_NOT_NULL`

File: src/parser/parser.yy [hunk 2]
```diff
@@ -672,7 +672,7 @@ expression_internal
     | expression_internal KW_IS_NOT_EMPTY {
         $$ = UnaryExpression::makeIsNotEmpty(qctx->objPool(), $1);
     }
-    | expression_internal EQ expression_internal {
+    | expression_internal ASSIGN expression_internal {
```
Comment (jievince) [inline src/parser/parser.yy:675]: emm, call `=`  `EQ` is better.

Comment (czpmango) [reply src/parser/parser.yy]: Fixed.

**Pull request closed without merging**